src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -485,6 +502,16 @@ export class MockHttpSocket extends MockSocket {
   ) => {
     this.shouldKeepAlive = shouldKeepAlive
 
+    if (rawHeaders === undefined) { 
+      rawHeaders = this.headers
+      this.headers = []
+    }
+
+    if (path === undefined) {
+      path = this.url
+      this.url = ''
+    }
```

Comment:
Do we need this?

Reply:
Yes, I believe it's necessary.
I haven't identified the exact root cause yet, but from what I observed, when there were too many headers and they weren't properly delivered, `path` value was also missing.